test/test_ff_utils.py

```diff
@@ -378,9 +526,90 @@ def test_authorized_request_integrated(integrated_ff):
     assert 'Bad status code' in str(exec_info.value)
 
 
-@pytest.mark.integrated
+def test_get_metadata_unit():
```

Comment:
There is a lot going on in this test, a high level description up here may be useful.

Reply:
I rearranged things slightly and added more comments.